client/components/TestQueue/index.jsx

```diff
@@ -22,19 +22,165 @@ import Actions from './Actions';
+  const [activeFilter, setActiveFilter] = useState(FILTER_KEYS.MANUAL);
   const { data, error, refetch } = useQuery(TEST_QUEUE_PAGE_QUERY, {
     fetchPolicy: 'cache-and-network'
   });
 
+  const hasBotRun = testPlanReport => {
+    return testPlanReport.draftTestPlanRuns?.some(({ tester }) => tester.isBot);
+  };
+
   const openDisclosuresRef = useRef({});
   const forceUpdate = useForceUpdate();
 
+  // Process data for hooks - must be done before any early returns
+  const processedData = useMemo(() => {
+    if (!data) return { testPlans: [], testers: [] };
+
+    const testPlanVersions = [];
+    data.testPlans.forEach(testPlan => {
+      // testPlan.directory is needed by ManageTestQueue
+      const populatedTestPlanVersions = testPlan.testPlanVersions.map(
+        testPlanVersion => ({
+          ...testPlanVersion,
+          testPlan: { directory: testPlan.directory }
+        })
+      );
+      testPlanVersions.push(...populatedTestPlanVersions);
+    });
+
+    // Remove any test plans or test plan versions without reports and sort
+    const sortTestPlanVersions = testPlanVersions => {
+      return [...testPlanVersions]
+        .filter(testPlanVersion => testPlanVersion.testPlanReports.length)
+        .sort((a, b) => {
+          return b.versionString.localeCompare(a.versionString);
+        })
+        .map(testPlanVersion => {
+          return {
+            ...testPlanVersion,
+            testPlanReports: sortTestPlanReports(
+              testPlanVersion.testPlanReports
+            )
+          };
+        });
+    };
+
+    const sortTestPlanReports = testPlanReports => {
+      return [...testPlanReports].sort((a, b) => {
+        if (a.at.name !== b.at.name) {
+          return a.at.name.localeCompare(b.at.name);
+        }
+        if (a.browser.name !== b.browser.name) {
+          return a.browser.name.localeCompare(b.browser.name);
+        }
+        const dateA = new Date(
+          (a.minimumAtVersion ?? a.exactAtVersion).releasedAt
+        );
+        const dateB = new Date(
+          (a.minimumAtVersion ?? a.exactAtVersion).releasedAt
+        );
+        return dateB - dateA;
+      });
+    };
+
+    const testPlans = data.testPlans
+      .filter(testPlan => {
+        for (const testPlanVersion of testPlan.testPlanVersions) {
+          if (testPlanVersion.testPlanReports.length) return true;
+        }
+      })
+      .map(testPlan => {
+        return {
+          ...testPlan,
+          testPlanVersions: sortTestPlanVersions(testPlan.testPlanVersions)
+        };
+      })
+      .sort((a, b) => {
+        return a.title.localeCompare(b.title);
+      });
+
+    const testers = data.users
+      .filter(user => user.roles.includes('TESTER'))
+      .sort((a, b) => a.username.localeCompare(b.username));
+
+    return { testPlans, testers };
+  }, [data]);
+
+  // Filter counts calculation
+  const { allCount, manualCount, automatedCount } = useMemo(() => {
```

Comment:
These values are incidental to the derivation of `filterOptions`, so they don't need to be memoized separately.

Reply:
Good catch! I think this was a leftover from when I was using those values elsewhere for some reason. Thanks! [190af56](https://github.com/w3c/aria-at-app/pull/1459/commits/190af56386b2f4e394c9a976657b699b82471388)